src/devices/HCSR501/HCSR501.cs

```diff
@@ -0,0 +1,86 @@
+﻿using System;
+using System.Devices.Gpio;
+
+namespace Iot.Device.HCSR501
+{
+    public class HCSR501ValueChangedEventArgs : EventArgs
+    {
+        public readonly PinValue PinValue;
+        public HCSR501ValueChangedEventArgs(PinValue value)
+        {
+            PinValue = value;
+        }
+    }
+
+    public class HCSR501 : IDisposable
+    {
+        private GpioPin sensor;
+        private readonly int _pinOut;
+
+        /// <summary>
+        /// Constructor
+        /// </summary>
+        /// <param name="pin">OUT Pin</param>
+        public HCSR501(int pin)
+        {
+            _pinOut = pin;
+        }
+
+        /// <summary>
+        /// Initialize the sensor
+        /// </summary>
+        public void Initialize()
+        {
+            GpioController controller = new GpioController(PinNumberingScheme.Gpio);
+
+            sensor = controller.OpenPin(_pinOut, PinMode.Input);
+
+            sensor.ValueChanged += Sensor_ValueChanged;
```

Comment:
This is outdated as well. The new way of doing this would be `controller.RegisterCallbackForPinValueChangedEvent(_pinOut, PinEventTypes.Rising, Sensor_ValueChanged)`